pkg/prometheus/promcfg_test.go

```diff
@@ -7452,7 +7491,50 @@ func TestScrapeConfigSpecConfigWithDigitalOceanSD(t *testing.T) {
 					},
 				},
 			},
-			golden: "ScrapeConfigSpecConfig_DigitalOceanSD_with_TLSConfig.golden",
+			version: "2.40.0",
+			golden:  "ScrapeConfigSpecConfig_DigitalOceanSD_with_TLSConfig.golden",
+		}, {
+			name: "digitalocean_sd_config_tls_with_unsupported_version",
```

Comment:
Why are we testing for TLS config when even DigitalOcean is not valid at that version?
This test case can be removed, it seems that the TLS configuration for DigitalOcean is supported across all versions that support DigitalOcean.